retriever/lib/engine.py

```diff
@@ -951,6 +969,20 @@ def extract_fixed_width(self, line):
         return values
 
 
+def set_csv_field_size():
+    """Set the CSV size limit based on the available resources"""
+    maxInt = sys.maxsize
+    decrement = True
+    while decrement:
+        decrement = False
+        try:
+            csv.field_size_limit(maxInt)
+        except OverflowError:
+            maxInt = int(maxInt / 10)
+            decrement = True
+    return maxInt
+
```

Comment:
I found this function difficult to understand. Can we: 1) add a bit more documentation; and 2) see if there is a more intuitive way to write this. The following feels a bit more intuitive to me since we set `decrement` to `True` and only change it when `csv.field_size_limit` works, updating the value used in the loop when it errors.

```python
def set_csv_field_size():
    """Set the CSV size limit based on the available resources"""
    maxInt = sys.maxsize
    decrement = True
    while decrement:
        try:
            csv.field_size_limit(maxInt)
            decrement = False
        except OverflowError:
            maxInt = int(maxInt / 10)
    return maxInt
```